crates/evm/coverage/src/inspector.rs

```diff
@@ -28,15 +43,50 @@ impl CoverageCollector {
     pub fn finish(self) -> HitMaps {
         self.maps
     }
+
+    #[inline]
+    fn get_or_insert_map(&mut self, interpreter: &mut Interpreter) -> &mut HitMap {
+        let hash = get_or_insert_contract_hash(interpreter);
+        if self.current_hash != *hash {
+            self.insert_map(interpreter);
+        }
+        unsafe { self.current_map.as_mut() }
+    }
+
+    #[cold]
+    #[inline(never)]
+    fn insert_map(&mut self, interpreter: &Interpreter) {
+        let Some(hash) = interpreter.contract.hash else { eof_panic() };
+        self.current_hash = hash;
+        self.current_map = self
+            .maps
+            .entry(hash)
+            .or_insert_with(|| HitMap::new(interpreter.contract.bytecode.original_bytes()))
+            .into();
```

Comment:
ah I see, 

I'd prefer the `NonNull::from` here explicitly and/or a one line doc